### PR TITLE
Nocodec amplifier+hdmi-in via I2S

### PR DESCRIPTION
**Pull request by CBala21**

### CODE REVIEW

**Comment by CBala21** [comment]
Hi Team,

Please do help to review and merge.

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (plbossart) [inline tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:163]: The naming of your HDMI capture devices is not really consistent with the playback case.
can I suggest:

HDMI IN 1
HDMI IN 2

**Review by plbossart — CHANGES_REQUESTED** [review]

File: tools/topology/topology1/CMakeLists.txt
Comment (CBala21) [reply tools/topology/topology1/CMakeLists.txt]: done. please check the latest.

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4]: Corrected. please check

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:26]: yes. We have products that doesn't have no SSP amp but need to support HDMI-IN capture.

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4]: Corrected. please check

**Comment by CBala21** [comment]
Hi plbossart,

Please check the latest patch which has been fixed the review comments and help to do get merge.

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (plbossart) [inline tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: that doesn't seem right, why would you receive the mclk and provide the bclk/fsync to the codec?

Is this intentional and if yes can you add a comment?

**Review by plbossart — CHANGES_REQUESTED** [review]
the mclk_in support is questionable at best. see comments below

File: tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4
Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: Yes. Amplifier case still its required and used.  For NO_AMP boards, this config wont be enabled.

Comment (plbossart) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: But you are using two different MCLK in config, with different values. How is this even supposed to work?

Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: Right. HDMI-In capture case really the MCLK is not used Since Lontium chip is master and DSP config not required. But in codec amplifier case it has been used our platform (TGL) MCLK (38.4 Mhz)configuration and value.

Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: Hi plbossart,

Hope I have clarified your question. Please help to proceed further if you have no more question.

Comment (plbossart) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: I don't see anything new to review @CBala21

Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: > I don't see anything new to review @CBala21

Could you provide your approval? waiting for one more approval.

Comment (plbossart) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: I am not convinced your solution works, the two MCLK configuration will conflict. To be clearer, the MCLK is unrelated to SSP, if you configure the MCLK in different ways for different SSPs, it's still the SAME MCLK hardware.

Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: Ok. I meant that HDMI-in capture case really NO MCLK required from platform Hardware and its generated and provided by external Lontium card which is converting HDMI inputs into I2S PCM and sending PCM samples via I2S pins. Hence what ever the values we mentioned in topology will not impact and not programmed by our DSP because we used config flag as "codec_master". Assumed in this case our DSP not consider MCLK value and wont generate and program MCLK. only bclk/fsync would be considered to support 32 bit and 48 Khzsamples.

also as my understanding, for codec amplifier case which ever the MCLK used is important to operate SSP amplifier. since we used flag "codec_slave" and DSP would program it.
we just followed the other codec amplifier topologies as a reference.

Please correct me if am wrong and suggest if any expectation on the values/parameters used on DAI CONFIG for HDMI-In capture. by the way the use case has been tested and delivered to TGL customers long time before and no issue found till now.

Comment (plbossart) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:197]: I don't think we have a macro to say 'NO MCLK'. @ranj063 may need to correct me here.

What you could do is use the same settings as for the amplifier, and add a comment that it's not used for the HDMI input. That way you wouldn't have any issue with conflicts on the MCLK setting.

We've done this before and used the same MCLK frequency/direction in other cases.

Comment (kv2019i) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:199]: @plbossart  @CBala21 Petér has been working on improving handling this case in https://github.com/thesofproject/sof/pull/7741 ... i.e. when SSP is not clock provider, we do not need to program MCLK. This was discovered in similar case where multiple SSPs were running with conflicting MCLKs  -> https://github.com/thesofproject/sof/issues/7803

Comment (CBala21) [reply tools/topology/topology1/sof-tgl-rt1308-hdmi-ssp.m4:199]: > I don't think we have a macro to say 'NO MCLK'. @ranj063 may need to correct me here.
> 
> What you could do is use the same settings as for the amplifier, and add a comment that it's not used for the HDMI input. That way you wouldn't have any issue with conflicts on the MCLK setting.
> 
> We've done this before and used the same MCLK frequency/direction in other cases.

Updated the file as per your suggestion. please check and provide the feedback.

**Review by plbossart — APPROVED** [review]
thanks @CBala21

**Pull request merged**
